scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -49,52 +65,141 @@ namespace <%= namespace %> {
             _LineItems = new CartLineItems();
         }
 
-        /// <summary>
-        /// Returns a link which can be used to check out this cart's line items via a web browser.
-        /// </summary>
-        /// <param name="note">can be used to create a special note for the merchant for this order</param>
-        /// \code
-        /// // Example that creates a checkout link and opens the url in the Browser
-        /// string linkToCheckout = ShopifyBuy.Client().Cart().GetWebCheckoutLink();
-        ///
-        /// Application.OpenURL(linkToCheckout);
-        /// \endcode
-        public string GetWebCheckoutLink(string note = null) {
-            StringBuilder url = new StringBuilder();
-            bool hasLineItem = false;
+        public void GetWebCheckoutLink(Action<string> success, Action<List<string>, string> failure) {
+            CheckoutSave((errors, httpError) => {
+                if (errors != null || httpError != null) {
+                    failure(errors, httpError);
+                } else {
+                    success(CurrentCheckout.webUrl());
+                }
+            });
+        }
+
+        private void CheckoutSave(Action<List<string>, string> callback) {
+            if (!IsCreated) {
+                CheckoutCreate(callback);
+            } else if(!IsSaved) {
+                CheckoutUpdate(callback);
+            } else {
+                callback(null, null);
+            }
+        }
+
+        private void CheckoutCreate(Action<List<string>, string> callback) {
+            MutationQuery query = new MutationQuery();
+
+            List<CheckoutLineItemInput> newLineItemInput = CartLineItems.ConvertToCheckoutLineItemInput(LineItems.New());
 
-            url.Append("http://");
-            url.Append(Client.Domain);
-            url.Append("/cart/");
+            DefaultQueries.checkout.Create(query, newLineItemInput);
 
-            foreach(CartLineItem lineItem in LineItems.All()) {
-                if (hasLineItem) {
-                    url.Append(",");
+            Client.Mutation(query, (Mutation response, List<string> errors, string httpError) => {
+                if (errors != null || httpError != null) {
```

Comment:
I'd prefer to keep it this way. Mainly because if there are `errors` or `httpErrors` we want to immediately fail and call the callback. For me it's actually easier to read since it's literally just one line in the if statement.